woocommerce/woocommerce-admin-test-helper
# Add REST API Filters to modify API responses

This PR adds a new tab that allows us to modify API responses using dot notation without changing code. It can come in handy when testing a new change that depends on an API response. 

### Testing Instruction

1. Install WooCommerce
2. Navigate to `WooComemrce -> Home` and click `Set up payments` task. We're going to modify the response of `/wc/v3/payment_gateways` endpoint.
3. Open inspector -> Network and select `payment_gateways?_locale=user`
4. Click `Preview` tab and confirm the first array element has `needs_setup` field and its value is `true`
5. Navigate to `Tools -> WCA Test Helper -> REST API Filters`
6. Click `New Filter` and enter the following values. Click `Create New Filter` button.

**Endpoint:** /wc/v3/payment_gateways
**Dot Notation:** 0.needs_setup
**Replacement:** false

7. Go back to the payment task and refresh the page.
8. Check the response again and confirm the `needs_setup` field is set to false.

### Screenshots

![Screen Shot 2022-04-14 at 2 01 29 PM](https://user-images.githubusercontent.com/4723145/163474467-b661777b-6e00-4962-956f-44652bd61942.jpg)
